src/dev_wasi_nn.erl

```diff
@@ -0,0 +1,324 @@
+%%% @doc A WASI-NN device implementation for HyperBEAM that provides AI inference
+%%% capabilities. This device supports loading models from Arweave transactions
+%%% and performing inference with session management for optimal performance.
+%%% Models are cached locally to avoid repeated downloads.
+-module(dev_wasi_nn).
+-export([info/1, info/3, infer/3]).
+-export([read_model_by_ID/1]).
+-include("include/hb.hrl").
+-include_lib("eunit/include/eunit.hrl").
+%% @doc Get device information and exported functions.
+%% Returns the list of functions that are exposed via the device API.
+%% This is used by the HyperBEAM runtime to determine which endpoints
+%% are available for this device.
+%%
+%% @param _ Ignored parameter.
+%% @returns A map containing the list of exported functions.
+info(_) -> 
+    #{ exports => [info, infer] }.
+
+%% @doc Provide HTTP info response about this device.
+%% Returns comprehensive information about the WASI-NN device including
+%% its capabilities, version, and API documentation. This endpoint helps
+%% users understand how to interact with the AI inference functionality.
+%%
+%% @param _Msg1 Ignored parameter.
+%% @param _Msg2 Ignored parameter.
+%% @param _Opts Ignored parameter.
+%% @returns {ok, InfoBody} containing device information and API documentation.
+info(_Msg1, _Msg2, _Opts) ->
+    InfoBody = #{
+        <<"description">> => <<"GPU device for handling LLM Inference">>,
+        <<"version">> => <<"1.0">>,
+        <<"api">> => #{
+            <<"infer">> => #{
+                <<"description">> => <<"LLM Inference">>,
+                <<"method">> => <<"GET or POST">>,
+                <<"required_params">> => #{
+                    <<"prompt">> => <<"Prompt for Infer">>,
+                    <<"model-id">> => <<"Arweave TX ID of the model file">>
```

Comment:
In general we use `TXID` rather than `TX ID`, but not a big deal -- there is some contention around this (some prefer `TxId`, `TxID` etc).